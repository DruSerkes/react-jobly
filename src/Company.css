.Company {
	margin-top: 3rem auto 0 auto;
	padding: 1rem 1rem;
}

.Company h2,
p {
	width: 50%;
	margin: 0 auto;
	text-align: center;
}
