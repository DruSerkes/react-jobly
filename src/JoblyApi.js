import axios from 'axios';

class JoblyApi {
	static async request(endpoint, paramsOrData = {}, verb = 'get') {
		// paramsOrData._token = // for now, hardcode token for "testing"
		// 	'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6InRlc' +
		// 	'3RpbmciLCJpc19hZG1pbiI6ZmFsc2UsImlhdCI6MTU1MzcwMzE1M30.' +
		// 	'COmFETEsTxN_VfIlgIKw0bYJLkvbRQNgO1XCSE8NZ0U';

		paramsOrData._token = localStorage.getItem('jobly-token') || null;

		console.debug('API Call:', endpoint, paramsOrData, verb);

		try {
			return (await axios({
				method                               : verb,
				url                                  : `http://localhost:3001/${endpoint}`,
				[verb === 'get' ? 'params' : 'data']: paramsOrData
			})).data;
			// axios sends query string data via the "params" key,
			// and request body data via the "data" key,
			// so the key we need depends on the HTTP verb
		} catch (err) {
			console.error('API Error:', err.response);
			let message = err.response.data.message;
			throw Array.isArray(message) ? message : [ message ];
		}
	}

	static async getCompany(handle) {
		let res = await this.request(`companies/${handle}`);
		return res.company;
	}

	static async getAllCompanies() {
		let res = await this.request(`companies`);
		return res.companies;
	}

	static async getAllJobs() {
		let res = await this.request('jobs');
		return res.jobs;
	}

	// This is going to need some debugging probably
	static async login(data) {
		let res = await this.request('login', data, 'post');
		console.log('res inside the API function: ', res);
		return res.token;
	}

	// So is this likely
	static async register(data) {
		let res = await this.request('users', data, 'post');
		console.log('res inside the API register function: ', res);
		return res.token;
	}
}

export default JoblyApi;
