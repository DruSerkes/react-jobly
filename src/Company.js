import React, { useState, useEffect } from 'react';
import JoblyApi from './JoblyApi';
import { useParams } from 'react-router-dom';
import Jobcard from './JobCard';
import './Company.css';

const Company = () => {
	const [ company, setCompany ] = useState(null);
	const { handle } = useParams();

	useEffect(() => {
		const getCompany = async () => {
			let company = await JoblyApi.getCompany(handle);
			setCompany({ ...company });
		};
		getCompany();
	}, []);

	if (!company) {
		return <h2>Loading company data</h2>;
	}

	return (
		<div className="Company">
			<h3>{company.name}</h3>
			<p>{company.description}</p>
			{company.jobs.map((job) => <Jobcard key={job.handle} job={job} />)}
		</div>
	);
};

export default Company;
