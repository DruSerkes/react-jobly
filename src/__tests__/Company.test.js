import React from 'react';
import { render } from '@testing-library/react';
import Company from '../Company';
import { MemoryRouter, Route } from 'react-router-dom';
import { useParams } from 'react-router';
import JoblyApi from '../JoblyApi';

describe('Company tests', () => {
	it('renders without crashing', () => {
		jest.mock('react-router', () => ({
			useParams : jest.fn().mockReturnValue({ handle: 'anderson-arias-and-morrow' })
		}));

		render(
			<MemoryRouter initialEntries={[ '/companies/anderson-arias-and-morrow' ]}>
				<Route exact path="/companies/:handle">
					<Company />
				</Route>
			</MemoryRouter>
		);
	});

	it('matches snapshot', () => {
		jest.mock('react-router', () => ({
			useParams : jest.fn().mockReturnValue({ handle: 'anderson-arias-and-morrow' })
		}));
		jest.mock('../JoblyApi', () => ({
			getCompany : jest.fn().mockReturnValue({ name: 'Anderson', description: 'testing them' })
		}));
		const { getByText, asFragment } = render(
			<MemoryRouter initialEntries={[ '/companies/anderson-arias-and-morrow' ]}>
				<Route exact path="/companies/:handle">
					<Company />
				</Route>
			</MemoryRouter>
		);
		expect(asFragment()).toMatchSnapshot();
		// expect(getByText('Anderson Arias and Morrow')).toBeInTheDocument();
	});
});
