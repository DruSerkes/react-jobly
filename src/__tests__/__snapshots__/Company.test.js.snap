// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Company tests matches snapshot 1`] = `
<DocumentFragment>
  <h2>
    Loading company data
  </h2>
</DocumentFragment>
`;
